### PR TITLE
Removing magic

### PR DESCRIPTION
**Pull request by pearce8**
Removing references to site-specific compiler wrappers.

### CODE REVIEW

**Comment by pearce8** [comment]
@rfhaque Please test all relevant benchmarks on the updated systems.

File: configs/ats4/auxiliary_software_files/packages.yaml
Comment (pearce8) [inline configs/ats4/auxiliary_software_files/packages.yaml:1]: Do we need this to be defined?  If yes, is there a non-tce version?

 git-lfs:
    externals:
    - spec: git-lfs@2.11.0
      prefix: /usr/tce

File: configs/ats2/variables.yaml
Comment (pearce8) [inline configs/ats2/variables.yaml:1]: @rfhaque Is there a non-tce location for lrun?  We are currently finding it in /usr/tcetmp/bin/lrun.

File: configs/ats4/auxiliary_software_files/compilers.yaml [hunk 1]
```diff
@@ -2,10 +2,10 @@ compilers:
...
     spec: cce@16.0.0-rocm5.5.1
     paths:
-      cc:  /usr/tce/packages/cce-tce/cce-16.0.0/bin/craycc
-      cxx:  /usr/tce/packages/cce-tce/cce-16.0.0/bin/crayCC
-      f77:  /usr/tce/packages/cce-tce/cce-16.0.0/bin/crayftn
-      fc:  /usr/tce/packages/cce-tce/cce-16.0.0/bin/crayftn
+      cc:  /opt/cray/pe/cce/16.0.0/bin/craycc
+      cxx:  /opt/cray/pe/cce/16.0.0/bin/crayCC
+      f77:  /opt/cray/pe/cce/16.0.0/bin/crayftn
+      fc:  /opt/cray/pe/cce/16.0.0/bin/crayftn
```
Comment (rfhaque) [inline configs/ats4/auxiliary_software_files/compilers.yaml]: @pearce8 amg2023 fails with the native cray compiler. It depends on workarounds in the magic wrapper

File: configs/ats4/auxiliary_software_files/packages.yaml [hunk 2]
```diff
@@ -91,8 +91,6 @@ packages:
     externals:
     - spec: git@2.31.1+tcltk
       prefix: /usr
-    - spec: git@2.29.1+tcltk
-      prefix: /usr/tce
```
Comment (becker33) [inline configs/ats4/auxiliary_software_files/packages.yaml]: This change seems orthogonal?

**Review by becker33 — COMMENTED** [review]
Looks good to me, commenting instead of approving because @rfhaque is doing manual testing.

File: configs/ats4/auxiliary_software_files/packages.yaml (hunk 2 above)
Comment (pearce8) [reply configs/ats4/auxiliary_software_files/packages.yaml]: yes, undoing.

**Review by rfhaque — CHANGES_REQUESTED** [review]
@pearce8 @scheibelp @becker33 Tested with the spack commit #8371bb4e. The following benchmarks have been tested
1. amg2023 (openmp/cts1, openmp/x86, openmp/ats2, openmp/ats4, rocm/ats4, cuda/ats2)
2. saxpy (openmp/ats4, rocm/ats4)
3. raja-perf (openmp/ats4, mpi-only/ats4, rocm/ats4)

**Comment by pearce8** [comment]
AMG fails on tioga:

    configure: error: linking to Fortran libraries from C fails

Moving work on this to a separate PR#89.

**Pull request merged**